src/plugins/posix/aio_queue.h

```diff
@@ -28,6 +29,7 @@ class nixlPosixBackendReqH;
 
 class aioQueue : public nixlPosixQueue {
     private:
+        std::mutex lock;                   // Protects all internal state from concurrent modification
```

Comment:
is nixlLock supposed to be used for plugins also?

Reply:
No. We need this mutex to protect queue internals when engine is executed concurrently on multiple threads because NIXL lock is taken in read mode by upper layer.

Reply:
After consulting with posix plugin developers it appears that the locking in those queues is redundant since the queues are per-transfer request and aren't shared. Removed that commit from the series.